src/Components/ExtractLevelsMenu.jsx

```diff
@@ -0,0 +1,184 @@
+
+import React, {useState, useEffect} from 'react'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import {makeStyles} from '@mui/styles'
+import {TooltipIconButton} from './Buttons'
+import useStore from '../store/useStore'
+import {getModelCenter} from '../utils/cutPlane'
+import {addHashParams, getHashParams} from '../utils/location'
+import {Vector3} from 'three'
+import {useLocation} from 'react-router-dom'
+import {removePlanes} from '../utils/cutPlane'
+import LevelsIcon from '../assets/2D_Icons/Levels.svg'
+import PlanViewIcon from '../assets/2D_Icons/PlanView.svg'
+import {extractHeight} from '../utils/extractHeight'
+
+/**
+ * BasicMenu used when there are several option behind UI button
+ * show/hide from the right of the screen.
+ *
+ * @param {Array} listOfOptions Title for the drawer
+ * @return {object} ItemPropertiesDrawer react component
+ */
+export default function ExtractLevelsMenu({listOfOptions, icon, title}) {
+  const [anchorEl, setAnchorEl] = useState(null)
+  const classes = useStyles()
+  const open = Boolean(anchorEl)
+
+  const model = useStore((state) => state.modelStore)
+  let [floorplanMenuItems, showExtractMenu] = useState([])
+
+
+  const PLANE_PREFIX = 'p'
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget)
+  }
+  const handleClose = () => {
+    setAnchorEl(null)
+  }
+  const viewer = useStore((state) => state.viewerStore)
+  const location = useLocation()
+  const createFloorplanPlane = (h1, h2) => {
+    console.log('Got here')
```

Comment:
Please remove and proofread rest of PR for these.

Reply:
Fixed